[qyaner] iP

qyaner opened the pull request:
# Duke 

> “Your mind is for having ideas, not holding them.” – David Allen

Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/qyaner/ip).
2. double-click it.
3. add your tasks.
4.let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
Level-0

Commit:
Level-1

Commit:
Level-2

Commit:
Level-3

Commit:
Level-4

Commit:
Level-5

Commit:
Level-6

Commit:
A-TextUiTesting

Commit:
separated classes into different files and added some comments

Commit:
Level-7

Commit:
Merge branch 'branch-Level-7'

Commit:
Level-8

Commit:
Merge branch 'branch-Level-8'

Commit:
A-MoreOOP

Commit:
Merge branch 'branch-A-MoreOOP'

Commit:
A-Packages

Commit:
Merge branch 'branch-A-Packages'

Commit:
Merge branch 'master' into add-gradle-support

Commit:
A-Gradle

Commit:
A-JUnit

Commit:
A-Jar

Commit:
A-JavaDoc

Commit:
A-CodingStandard

Commit:
Level-9

Commit:
Merge branch 'branch-A-JavaDoc'

Commit:
Merge branch 'branch-A-CodingStandard'

Commit:
Merge branch 'branch-Level-9'

qyaner committed:
Update README.md

sk2001git commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,64 @@
+import duke.*;
```
I think you should not use the * operator for imports, it is generally recommended that you import every package individually, i think i see this at some other class files as well

sk2001git commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,99 @@
+ * @author Qin Yan Er
+ */
+public class Deadline extends Task {
+    protected String by;
```
I think a noun would be better to describe the String "by"!

sk2001git replied:
i think a possible suggestion could be  "startTime" ?

sk2001git commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,135 @@
+     * @param list the list of tasks.
+     * @param number the task number to be unmarked.
+     */
+    public void unMark(TaskList list, int number) {
```
I think this should be unmark since its is a word ?

sk2001git commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,135 @@
+     * @param list the list of tasks.
+     * @param newEvent the new event to be added.
+     */
+    public void addEvent(TaskList list, Event newEvent) {
+        System.out.println("Got it. I've added this task:");
+        System.out.println(newEvent.toString());
+        System.out.println("Now you have " + list.getSize() + " tasks in the list.");
+    }
+
```
In general, I think that you have done quite well in documentation, naming of methods and variable names, naming seem intuitive and well ordered, well done! 👍 😄

sk2001git reviewed (commented):
LGTM. I think there was some minor coding standard violations and naming problems, so small number of nits however naming was generally well done! 👍

qyaner committed:
Update README.md

Commit:
Level-10: Created GUI for Duke

Commit:
put the new classes into duke package

Commit:
Merge branch 'branch-Level-10'

Commit:
Merge branch 'master' of https://github.com/qyaner/ip

Commit:
update README.md

Commit:
Add assertions in various classes

Classes have important assumptions that may not hold at various points in the code.

Let's add assertions so that such assumptions can be defined and so the runtime can verify them. For example, we can check if the input is null by using assertions.

Commit:
Add and improve on javadocs for all public classes and methods

Commit:
Parser class: separate parse method into shorter methods

Parse method was too long and may have affected code readability.

Let's refactor the parse method by breaking it down into smaller methods. Each new method is responsible for parsing and executing a specific type of command.

qyaner committed:
Merge pull request #1 from qyaner/branch-A-CodeQuality

Improve code quality

qyaner committed:
Merge branch 'master' into branch-A-Assertions

qyaner committed:
Merge pull request #2 from qyaner/branch-A-Assertions

Add assertions in various classes

Commit:
Parser class: Add javadocs to new methods

Commit:
TaskList class: Implement duplicate task detection in add method

Commit:
Merge branch 'branch-extensions'

Commit:
ParserTest class: Update to test the changed Parser class

Commit:
Add Ui.png

qyaner committed:
Update README.md

qyaner committed:
Update README.md

Commit:
Add some assertions and update ParserTest class

Commit:
Merge branch 'master' of https://github.com/qyaner/ip

Commit:
delete commented out code
